### PR TITLE
Deploy full oauth functionality

### LINKED ISSUE #93: Integrate with ebay official auth module
use the below library to generate the access token.

https://github.com/eBay/ebay-oauth-nodejs-client

### PR DESCRIPTION
**Pull request by TotallyNotChase**
## Goal of the pull request:
* [ ] Documentation
* [ ] Bugfix
* [x] Feature (New!)
* [ ] Enhancement

## Description
Sets up oauth for both application access tokens (client credentials) and user access tokens (authorization grant). Closes #93 

## Changelog
* Add `getUserAuthorizationUrl`, `getUserTokenByCode`, and `getUserTokenByRefresh` for getting user access tokens
* Add `setUserAccessToken` and `setAppAccessToken`, for setting user token and app access token respectively. These are designed to take the ebay api response as a parameter and set them accordingly
* Add `oauthEndpoint` for both `SANDBOX` and `PROD` to constants, this is used for generating the user consent url
* Rename `src/common-utils/index/getAccessToken` -> `src/index/getAccessToken`, with a few minor changes. Namely, explicitly set `this.options.data` instead of using `this.options.body`. This is done to be streamlined across all the token getter functions. `getUserTokenByCode` must use a custom `this.options.data` as it requires a `code`.

**Note**: This kind of makes `this.options.body.grant_type` useless, because each grant type has its own function but this will not break front end user functionality. In time, I suggest the `options.body` be completely removed and the `scopes` be passed in to their respective token getters as a parameter, hence achieving Single Responsibility Principle.
* Rename  `src/common-utils/index/setAccessToken`  -> `src/index/setAppAccessToken`, with a few minor changes. The parameter is now the full response returned by the ebay server and instead of being stored in `options.access_token`, it's now stored in `options.appAccessToken`
* Rename `this.options.access_token` -> `this.options.appAccessToken`
* Add `getUserToken` demo

## Things to keep in mind during dev
* The endpoints that require an application access token (client credentials) should use `this.options.appAccessToken` (previously `this.options.access_token`)
* The endpoints that require a user access token (authorization grant) should use `this.options.userAccessToken` (previously unavailable)
* The refresh token for user access token is stored in `this.options.refreshToken`
* When a user is redirected after granting consent in the url returned by `getUserAuthorizationUrl`, the code present in the redirected url is **url encoded**. It must be **url decoded** before being passed into `getUserTokenByCode`

### CODE REVIEW

**Comment by TotallyNotChase** [comment]
@pajaydev did you get a chance at reviewing the changes?

File: src/index.js [hunk 1]
```diff
@@ -1,45 +1,200 @@
...
+/**
+ * Assign application access token returned from client credentials workflow (i.e getAccessToken)
+ *
+ * @param appAccessToken appAccessToken obj returned from getApplicationToken
+*/
+const setAppAccessToken = function (appAccessToken) {
+    if (!appAccessToken.token_type === 'Application Access Token') throw new Error('appAccessToken is either missing or invalid');
+    this.options.appAccessToken = appAccessToken.access_token;
+};
+
```
Comment (pajaydev) [inline src/index.js:190]: @TotallyNotChase  Shall we keep it in a new file as these methods are related to access tokens like file `credentials.js`?.

**Comment by pajaydev** [comment]
@TotallyNotChase  Sorry for the delay, I was caught up with some other work. I gave a small comment, kindly let me know if you are busy I will merge this and do it as part of cleanup. Thanks again for your patience 👍

**Comment by TotallyNotChase** [comment]
@pajaydev The changes are now migrated to `src/credentials.js`

**Comment by TotallyNotChase** [comment]
@pajaydev sounds good! thanks!

**Comment by emporioreale** [comment]
How to properly use that feature?
Please post the working code to get an user access token by a refresh token.
The demo code can't work because the refresh token parameter of the getUserTokenByRefresh method is missing.

**Pull request merged**